From 2c1421b075728cd5189d4dbfbac6afa1e32777dc Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 17 Mar 2025 09:06:57 +0000
Subject: [PATCH 1/5] linux: Provide appstream metainfo file

This file is used by linux software centers to display and promote
the application.
---
 .../com.mindforger.mindforger.metainfo.xml    | 36 +++++++++++++++++++
 mindforger.pro                                |  5 +++
 2 files changed, 41 insertions(+)
 create mode 100644 app/resources/linux/com.mindforger.mindforger.metainfo.xml

diff --git a/app/resources/linux/com.mindforger.mindforger.metainfo.xml b/app/resources/linux/com.mindforger.mindforger.metainfo.xml
new file mode 100644
index 00000000..8830ce03
--- /dev/null
+++ b/app/resources/linux/com.mindforger.mindforger.metainfo.xml
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.mindforger.mindforger</id>
+
+  <name>Mindforger</name>
+  <summary>Thinking notebook and Markdown editor. </summary>
+
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0-or-later</project_license>
+
+  <recommends>
+    <display_length compare="ge">768</display_length>
+  </recommends>
+
+  <description>
+    <p>
+      Are you drowning in information, but starving for knowledge?
+    </p>
+    <p>
+      Where do you keep your private remarks like ideas, personal plans, exam preparation notes, gift tips, how-tos, dreams, business visions, finance strategies, meeting minutes and auto coaching notes? Loads of documents, sketches and remarks spread around the file system, cloud, web and Post-it notes? Are you afraid of your knowledge privacy? Are you able to find particular remarks once you create them? Do you know how are the remarks mutually related when you browse, read or write them? No?
+    </p>
+  </description>
+
+  <launchable type="desktop-id">mindforger.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://www.mindforger.com/images/screenshots/tayr.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://www.mindforger.com/images/screenshots/knowledge-graph-navigator-light.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://www.mindforger.com/images/screenshots/tayw.png</image>
+    </screenshot>
+  </screenshots>
+</component>
diff --git a/mindforger.pro b/mindforger.pro
index 8bb93ec1..1510fcdd 100644
--- a/mindforger.pro
+++ b/mindforger.pro
@@ -68,4 +68,9 @@ shortcutfiles.files += app/resources/gnome-shell/mindforger.desktop
 shortcutfiles.path = $$PREFIX/share/applications/
 INSTALLS += shortcutfiles
 
+# Appstream Metainfo for linux software centers
+appstreammetainfofiles.files += app/resources/linux/com.mindforger.mindforger.metainfo.xml
+appstreammetainfofiles.path = $$PREFIX/share/metainfo/
+INSTALLS += appstreammetainfofiles
+
 # eof

From 8fccc5fd641705802ac685778611af66c8b4844b Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 17 Mar 2025 09:10:17 +0000
Subject: [PATCH 2/5] linux: Reorganize .desktop file

The handling of the .desktop file applies to every linux desktop
environment rather than gnome-shell specifically.
---
 app/resources/{gnome-shell => linux}/howto.md           | 0
 app/resources/{gnome-shell => linux}/mindforger.desktop | 0
 mindforger.pro                                          | 7 +++----
 3 files changed, 3 insertions(+), 4 deletions(-)
 rename app/resources/{gnome-shell => linux}/howto.md (100%)
 rename app/resources/{gnome-shell => linux}/mindforger.desktop (100%)

diff --git a/app/resources/gnome-shell/howto.md b/app/resources/linux/howto.md
similarity index 100%
rename from app/resources/gnome-shell/howto.md
rename to app/resources/linux/howto.md
diff --git a/app/resources/gnome-shell/mindforger.desktop b/app/resources/linux/mindforger.desktop
similarity index 100%
rename from app/resources/gnome-shell/mindforger.desktop
rename to app/resources/linux/mindforger.desktop
diff --git a/mindforger.pro b/mindforger.pro
index 1510fcdd..baebe302 100644
--- a/mindforger.pro
+++ b/mindforger.pro
@@ -63,10 +63,9 @@ iconfiles.files += app/resources/icons/*
 iconfiles.path = $$PREFIX/share/icons/mindforger/
 INSTALLS += iconfiles
 
-# experiment w/ file
-shortcutfiles.files += app/resources/gnome-shell/mindforger.desktop
-shortcutfiles.path = $$PREFIX/share/applications/
-INSTALLS += shortcutfiles
+linuxdesktopfiles.files += app/resources/linux/mindforger.desktop
+linuxdesktopfiles.path = $$PREFIX/share/applications/
+INSTALLS += linuxdesktopfiles
 
 # Appstream Metainfo for linux software centers
 appstreammetainfofiles.files += app/resources/linux/com.mindforger.mindforger.metainfo.xml

From 0ca57da95a93994982f9f4985054318a0bfe76f9 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 17 Mar 2025 09:28:56 +0000
Subject: [PATCH 3/5] linux: Install icons to standard paths

Some applications such as software centers will only look in
standard linux icon paths to resolve an icon.
---
 mindforger.pro | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/mindforger.pro b/mindforger.pro
index baebe302..fdb12be7 100644
--- a/mindforger.pro
+++ b/mindforger.pro
@@ -63,6 +63,16 @@ iconfiles.files += app/resources/icons/*
 iconfiles.path = $$PREFIX/share/icons/mindforger/
 INSTALLS += iconfiles
 
+# Standard location for svg icon
+linuxsvg.files += app/resources/icons/mindforger.svg
+linuxsvg.path = $$PREFIX/share/icons/hicolor/scalable/apps/
+INSTALLS += linuxsvg
+
+# Standard location for png icon
+linuxpng.files += app/resources/icons/mindforger128x128.png
+linuxpng.path = $$PREFIX/share/icons/hicolor/128x128/apps/
+INSTALLS += linuxpng
+
 linuxdesktopfiles.files += app/resources/linux/mindforger.desktop
 linuxdesktopfiles.path = $$PREFIX/share/applications/
 INSTALLS += linuxdesktopfiles

From f7f72eaed9613b1706df67e45ed7ebb0104487ec Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 17 Mar 2025 09:30:10 +0000
Subject: [PATCH 4/5] linux: Don't hardcode locations in .desktop file

With assets installed to standard paths this can be resolved
automatically.
---
 app/resources/linux/mindforger.desktop | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/app/resources/linux/mindforger.desktop b/app/resources/linux/mindforger.desktop
index ddcfed9a..5e5659f0 100644
--- a/app/resources/linux/mindforger.desktop
+++ b/app/resources/linux/mindforger.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=MindForger
-Exec=/usr/bin/mindforger
-Icon=/usr/share/icons/mindforger/mindforger.png
+Exec=mindforger
+Icon=mindforger
 Type=Application
 Categories=Office;

From 7179746007554b8bb743bb10ca5338988a28e15f Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 17 Mar 2025 09:58:26 +0000
Subject: [PATCH 5/5] linux: Expand appstream metainfo info & pass appstreamcli
 validate

---
 .../com.mindforger.mindforger.metainfo.xml    | 22 ++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/app/resources/linux/com.mindforger.mindforger.metainfo.xml b/app/resources/linux/com.mindforger.mindforger.metainfo.xml
index 8830ce03..f7c1ec64 100644
--- a/app/resources/linux/com.mindforger.mindforger.metainfo.xml
+++ b/app/resources/linux/com.mindforger.mindforger.metainfo.xml
@@ -3,10 +3,19 @@
   <id>com.mindforger.mindforger</id>
 
   <name>Mindforger</name>
-  <summary>Thinking notebook and Markdown editor. </summary>
+  <summary>Thinking notebook and Markdown editor</summary>
 
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-2.0-or-later</project_license>
+  <developer id="com.mindforger">
+    <name>Martin Dvorak</name>
+  </developer>
+
+  <url type="homepage">https://www.mindforger.com/</url>
+  <url type="bugtracker">https://github.com/dvorka/mindforger/issues</url>
+  <url type="faq">https://github.com/dvorka/mindforger/wiki/FAQs</url>
+  <url type="help">https://github.com/dvorka/mindforger/wiki</url>
+  <url type="vcs-browser">https://github.com/dvorka/mindforger</url>
 
   <recommends>
     <display_length compare="ge">768</display_length>
@@ -21,6 +30,8 @@
     </p>
   </description>
 
+  <content_rating type="oars-1.0"/>
+
   <launchable type="desktop-id">mindforger.desktop</launchable>
   <screenshots>
     <screenshot type="default">
@@ -32,5 +43,14 @@
     <screenshot>
       <image>https://www.mindforger.com/images/screenshots/tayw.png</image>
     </screenshot>
+    <screenshot>
+      <image>https://www.mindforger.com/images/screenshots/markdown-images.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://www.mindforger.com/images/screenshots/hoisting.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://www.mindforger.com/images/screenshots/i18n.png</image>
+    </screenshot>
   </screenshots>
 </component>
